test/integration/com/redhat/theses/UniversityControllerIntegrationSpec.groovy

```diff
@@ -0,0 +1,111 @@
+package com.redhat.theses
+
+import com.redhat.theses.auth.User
+import com.redhat.theses.auth.Role
+import org.codehaus.groovy.grails.plugins.springsecurity.SpringSecurityUtils
+import grails.plugin.spock.IntegrationSpec
+import spock.lang.Shared
+import spock.lang.Unroll
+import grails.test.mixin.*
+
+class UniversityControllerIntegrationSpec extends IntegrationSpec {
+    @Shared controller = new UniversityController()
+
+    def setupSpec() {
+    }
+
+    def cleanup() {
+    }
+
+    void "Can retrieve university by ID, should succeed"() {
+        when: "retrieving university by ID"
+            def model = controller.show(1)
+        then: "should retrieve correct university"
+            model?.universityInstance?.name == "First"
+            model?.universityInstance?.acronym == "F"
+    }
+    
+    void "Can retrieve university by ID, should fail"() {
+        when: "trying to retrieve university by ID"
+            def model = controller.show(4)
+        then: "should retrieve nothing and redirect to list of universities"
+            model == null
+            controller.flash.message != null
+            controller.response?.redirectUrl.endsWith("list")
+    }
+    
+    void "Can edit university, should fail"() {
```

Comment:
Not like this..  The point of this BDD style naming is to use plain English as much as possible.

"Should not be able to edit university" 

or even better

"Can not edit university as regular user"